.github/workflows/main.yml

```diff
@@ -50,3 +80,31 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           PR_NUMBER: ${{ github.event.pull_request.number }}
+  
+  Generate_Docs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+
+      - name: Install pre-commit
+        run: |
+          python -m pip install --upgrade pip
+          pip install pre-commit
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '18.8.0'
+
+      - name: Install action-docs
+        run: npm install action-docs
+
+      - name: Run pre-commit for GitHub Actions Docs
+        run: pre-commit run -a github-action-docs
+      
```

Comment:
Any reason to run this separately instead of as part of the other pre-commit set up?

Reply:
No, I am replicating the existing behavior

Reply:
Cool, makes sense